frame/support/src/traits.rs

```diff
@@ -53,15 +53,15 @@ impl<T: Default> Get<T> for () {
 /// A trait for querying whether a type can be said to statically "contain" a value. Similar
 /// in nature to `Get`, except it is designed to be lazy rather than active (you can't ask it to
 /// enumerate all values that it contains) and work for multiple values rather than just one.
-pub trait Contains<T> {
+pub trait Contains<T: Ord> {
 	/// Return `true` if this "contains" the given value `t`.
-	fn contains(t: &T) -> bool;
-}
+	fn contains(t: &T) -> bool { Self::sorted_members().binary_search(t).is_ok() }
 
-impl<V: PartialEq, T: Get<V>> Contains<V> for T {
```

Comment:
this won't work if we also have specialisation for a module as we do now with the treasury module, hence the need for these changes and `ord_parameter_types`.